Add String.is_substring_of to check for consecutive match

dreamsComeTrue opened the pull request:
Non-breaking change - I've extended two methods in core ```ustring``` of additional parameter, though its default value retains current behaviour.

![image](https://user-images.githubusercontent.com/1110337/76699895-102fc780-66b2-11ea-819f-122df4fcb79a.png)


Fixes: #37063

dreamsComeTrue committed:
Add String.is_substring_of to check for consecutive match.

Fixes: #37063

Dragoncraft89 commented:
As I've commented in #37063, I think it would be better to create a new method String.is_substring_of than adding this parameter, as this would be more clear, what the method is supposed to do.

dreamsComeTrue commented:
Well, you may be right - although it expands String API a bit, it may really be a nice tradeoff compared to using cryptic-params. I've modified my PR, and updated docs - now it is clear what is what :)

![image](https://user-images.githubusercontent.com/1110337/76702738-8e996300-66cc-11ea-9d66-5ce71aaae378.png)

YuriSizov commented:
The original issue has been addressed via documentation changes, and there doesn't seem to be much interest in this addition. I'm bumping it to 4.x.

Ivorforce commented:
Hi, sorry to zombie an old thread! 
I'm just going through some old PRs, and I'm surprised it hasn't been mentioned here that the functionality you're looking for has been available in `string.find(other_string)` all along! So adding a duplicate function `is_substring_of` probably isn't the best idea. But it might be worth linking to `find` from `is_subsequence_of`!
